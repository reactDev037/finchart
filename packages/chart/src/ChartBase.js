import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Construct a Chart wrapper
 */
export class Component extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    transform: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ]),
  };

  static defaultProps = {
    children: null,
    transform: null,
  };

  render() {
    const { className, width, height, transform, children } = this.props;
    return (
      <svg {...{ className, width, height, transform }}>
        <rect className="background" {...{ width, height }} />
        {children}
      </svg>
    );
  }
}

// TODO: update the styled approach to make it more readable
export default styled(Component)`
  display: block;
  padding: ${prop => prop.padding || 60}px;

  > rect.background {
    fill: #e9e9eb;
    fill-opacity: 0.4;
  }
`;
