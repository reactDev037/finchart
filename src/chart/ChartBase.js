import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';

/**
 * Construct a Chart wrapper
 */
export class Component extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ]),
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const {
      className,
      width,
      height,
      children,
    } = this.props;
    return (
      <svg {...{ className, width, height }}>
        <rect className="background" {...{ width, height }} />
        {children}
      </svg>
    );
  }
}

// TODO: update the styled approach to make it more readable
export default styled(Component)`
  display: block;
  padding: ${prop => prop.padding || 60}px;

  > rect.background {
    fill: lightblue;
    fill-opacity: 0.4;
  }
`;
