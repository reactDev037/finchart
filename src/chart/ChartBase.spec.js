import React from 'react';

import ChartBase from './ChartBase';

describe('<ChartBase />', () => {
  it('renders', () => {
    const chartConfig = {
      width: 960,
      height: 320,
    };
    const wrapper = render(<ChartBase {...chartConfig} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children elements', () => {

  });
});
