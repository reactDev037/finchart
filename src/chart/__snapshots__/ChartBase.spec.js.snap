// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartBase /> renders 1`] = `
<svg
  class="sc-bdVaJa gtUTv"
  height="320"
  width="960"
>
  <rect
    class="background"
    height="320"
    width="960"
  />
</svg>
`;
